ang/test/civicase-base/directives/send-to-api-on-change.directive.spec.js

```diff
@@ -0,0 +1,86 @@
+/* eslint-env jasmine */
+
+(($, _) => {
+  describe('civicaseInlineDatepicker', () => {
+    let $compile, $q, $rootScope, $scope, civicaseCrmApi, crmStatus, element;
+
+    beforeEach(module('civicase-base', 'civicase.data', ($provide) => {
+      civicaseCrmApi = jasmine.createSpy('civicaseCrmApi');
+      crmStatus = jasmine.createSpy('crmStatus');
+
+      $provide.value('civicaseCrmApi', civicaseCrmApi);
+      $provide.value('crmStatus', crmStatus);
+    }));
+
+    beforeEach(inject((_$compile_, _$q_, _$rootScope_,
+      _civicaseCrmApi_) => {
+      $compile = _$compile_;
+      $q = _$q_;
+      $rootScope = _$rootScope_;
+      $scope = $rootScope.$new();
+      civicaseCrmApi = _civicaseCrmApi_;
+    }));
+
+    beforeEach(() => {
+      $scope.handleDataSent = jasmine.createSpy('handleDataSent');
+      $scope.apiData = ['Contact', 'create', { id: 1, display_name: 'abc' }];
+      $scope.myModel = 123;
+
+      crmStatus.and.returnValue($q.resolve());
+      civicaseCrmApi.and.returnValue($q.resolve());
+      initDirective();
+    });
+
+    describe('when the input changes', () => {
+      beforeEach(() => {
+        element.val(456);
```

Comment:
@deb1990 this value is not sent to the Backend. The value sent to the backend is anything that is defined in the `data-api-data` attribute. If we want to send this value we need to do something like this:

```html
        <input
          civicase-send-to-api-on-change
          data-api-data="getApiData()"
          ng-model="myModel"
          type="text"
        />
```

```js
$scope.getApiData = function () {
  return ['Entity', 'action', { value: $scope.myModel }];
};
```

The reason for this is that if we try to use the value directly we'd need to define more attributes on the input so we know which entity, action, and field needs to be updated:

```html
<input
  civicase-send-to-api-on-change
  data-entity="Entity"
  data-action="action"
  data-field="value"
  ng-model="myModel"
/>
```

Apart from being more verbose, this makes the directive less flexible because we are only limited to a single request. If we pass the API params through `data-api-data` we can make multiple requests. Something that is needed in our case since we need to update the date but also create an activity.